### DISCUSSION TITLE
Videotexture Playback stuttering

### DISCUSSION THREAD

**Discussion by wirmachenbunt** [cables-gl/cables_docs#733] [Answers & Questions]
is there any way to make videotexture playback smooth ? right now it's not really usable this way.
https://cables.gl/p/iKQcXa

and can also be seen in the demo patch with the clouds video
https://cables.gl/p/E0hWmi

**Comment by pandrr** [comment]
example can't be opened in editor, so can't really look at it

the example is a stop motion video, it is not fluid, yes its a bad example, i dont have anything better with a license we can use

what codec is this video ? is seems to be transparent, thats not a standard feature of browser videos, so thats probably slow ?

**Reply by wirmachenbunt** [reply]
the title was misleading, its a standard mp4 file
https://cables.gl/p/iKQcXa

when playing the video as videoelement and not as texture, it plays smooth

**Comment by pandrr** [comment]
https://dev.cables.gl/edit/ylBr2b

this is 60fps 1080p it plays fine for me

**Reply by wirmachenbunt** [reply]
not the best video to see if it stutters but this stutters as well, on multiple PC's with big GPU's

**Reply by wirmachenbunt** [reply]
https://github.com/cables-gl/cables_docs/assets/1138673/7a97e70f-3e52-42e0-aaf2-8fa27243af2f

**Reply by wirmachenbunt** [reply]
on the mac with its high refresh rate, its probably not as visible compared to a 60hz display. comparing your video in a videoplayer software to the playback as a texture in cables, the difference is very much visible

**Comment by pandrr** [comment]
i dont know 

maybe its the same as this discussion:

https://github.com/mrdoob/three.js/issues/13379
https://web.dev/articles/requestvideoframecallback-rvfc

that solution does not work in ff

**Comment by pandrr** [comment]
on dev.cables.gl the op now uses requestVideoFrameCallback if available and ignores the input fps number (output port "auto fps" is then true)

please check if this gives you better results.
